packages/core/src/types.ts

```diff
@@ -195,6 +199,27 @@ export interface RemoteComponent<
   ): void | Promise<void>;
 }
 
+export interface RemoteFragment<
+  Root extends RemoteRoot<any, any> = RemoteRoot<any, any>
+> {
+  readonly kind: typeof KIND_FRAGMENT;
+  readonly id: string;
+  readonly children: readonly AllowedChildren<ExtractChildren<any>, Root>[];
+  readonly root: Root;
+  readonly top: RemoteComponent<any, Root> | Root | null;
+  readonly parent: RemoteComponent<any, Root> | Root | null;
+  appendChild(
+    child: AllowedChildren<ExtractChildren<any>, Root, true>,
```

Comment:
Should we allow calling `root.appendChild` and `component.appendChild` with a fragment? It would make vanilla more powerful. You can create a fragment and do dynamic operations on it and then be able to append it to any node which allows flexibility in how the component tree is built.